From a979dcb1c8c661901dcfc6d75fffb35185abd4f2 Mon Sep 17 00:00:00 2001
From: Srinivasa Reddy Vundela <vsr@cloudera.com>
Date: Tue, 1 Dec 2015 15:04:36 -0800
Subject: [PATCH 194/201] CLOUDERA-BUILD CDH-27093 Checking for the required
 hadoop libraries before starting the spark

(cherry picked from commit 2468d55923357f68eed8f5e9d41cbda4d9160950)
---
 bin/spark-class |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/bin/spark-class b/bin/spark-class
index 5d964ba..fe6ff4a 100755
--- a/bin/spark-class
+++ b/bin/spark-class
@@ -80,8 +80,24 @@ fi
 # The launcher library will print arguments separated by a NULL character, to allow arguments with
 # characters that would be otherwise interpreted by the shell. Read that in a while loop, populating
 # an array that will be used to exec the final command.
+CP_INDEX=-1
+INDEX=0
 CMD=()
 while IFS= read -d '' -r ARG; do
+  if [ "$ARG" = "-cp" ]; then
+    CP_INDEX=$((INDEX+1))
+  fi
   CMD+=("$ARG")
+  (( INDEX++ ))
 done < <("$RUNNER" -cp "$LAUNCH_CLASSPATH" org.apache.spark.launcher.Main "$@")
+
+# Check for the required libraries before starting spark. Exit if any of the required libraries
+# are missing.
+if [[ ! "${CMD[$CP_INDEX]}" =~ "hadoop-common" ]]; then
+  echo "Required Hadoop libraries are missing in classpath." 1>&2
+  echo "This could happen if gateway roles are not setup on the node or configuration files are missing." 1>&2
+  echo "Please check Spark and Hadoop configurations, setup gateway roles on this node and try again." 1>&2
+  exit 1
+fi
+
 exec "${CMD[@]}"
-- 
1.7.9.5

